pulp_container/app/models.py

```diff
@@ -188,18 +195,66 @@ class Meta:
         unique_together = (("name", "tagged_manifest"),)
 
 
-class ContainerNamespace(BaseModel, AutoAddObjPermsMixin, AutoDeleteObjPermsMixin):
+class ContainerNamespace(BaseModel, AutoAddObjPermsMixin):
     """
     Namespace for the container registry.
     """
 
     name = models.CharField(max_length=255, db_index=True)
     ACCESS_POLICY_VIEWSET_NAME = "pulp_container/namespaces"
 
+    def create_namespace_group(self, permissions, parameters):
+        """
+        Creates a namespace group and optionally adds the current user to it.
+
+        The parameters are specified as a dictionary with the following keys:
+
+        "group_type" - the type of group - owners, collaborators, or consumers
+        "add_user_to_group" - a boolean that specifies if the current user should be added to the
+                              group.
+
+        The permissions are object level permissions assigned to the group.
+        """
+
+        group_type = parameters["group_type"]
+        add_user_to_group = parameters["add_user_to_group"]
+        group = Group.objects.create(
+            name="{}.{}.{}".format("container.namespace", group_type, self.name)
+        )
+        current_user = get_current_authenticated_user()
+        owners_group = Group.objects.get(
+            name="{}.{}".format("container.namespace.owners", self.name)
+        )
+        assign_perm("auth.change_group", owners_group, group)
+        assign_perm("auth.view_group", owners_group, group)
+        if add_user_to_group:
+            current_user.groups.add(group)
+        self.add_for_groups(permissions, group.name)
+
+    @hook("before_delete")
+    def delete_groups_and_user_obj_perms(self):
```

Comment:
@dkliban I have added the AutoDeleteObjPermMixin and it seems like hook in the core is called fist, and then this one. Shall we add this mixin so we are sure we are also removing `GroupObjectPermission.`?

Reply:
Django lifecycle does not have any docs about decorating two methods with the same 'when' condition. I prefer to just add the         `GroupObjectPermission.objects.filter(object_pk=self.pk).delete()` to this method.